script/makeinstaller.sh

```diff
@@ -262,17 +280,51 @@ StrCmp \$0 \${ffmpeg} "" +2
 [% endblock %]
 EOF
 
-echo "Building Python 3 installer" 1>&2
-
 # copy the streamlinkrc file to the build dir, we cannot use the Include.files property in the config file
 # because those files will always overwrite, and for a config file we do not want to overwrite
-cp "win32/streamlinkrc" "${files_dir}/streamlinkrc"
+cp "${ROOT}/win32/streamlinkrc" "${files_dir}/streamlinkrc"
+
+
+# download binary assets like ffmpeg and rtmpdump from the streamlink assets repo
+# parse the data.json manifest, validate archives and copy specific files to their destination
+log "Fetching assets data from \"${STREAMLINK_ASSETS_REPO}\" (${STREAMLINK_ASSETS_RELEASE})"
+assets_release_data=$(curl -s --fail \
+    -H 'Accept: application/vnd.github.v3+json' \
+    -H "User-Agent: ${GITHUB_REPOSITORY:-"streamlink/streamlink"}" \
+    "https://api.github.com/repos/${STREAMLINK_ASSETS_REPO}/releases/${STREAMLINK_ASSETS_RELEASE}" \
+    || err "Could not fetch release data"
+)
+assets_release_commit=$(echo "${assets_release_data}" | jq -r ".target_commitish")
+assets_data=$(curl -s --fail \
+    -H "User-Agent: ${GITHUB_REPOSITORY:-"streamlink/streamlink"}" \
+    "https://raw.githubusercontent.com/${STREAMLINK_ASSETS_REPO}/${assets_release_commit}/data.json" \
+    || err "Could not fetch manifest data"
+)
+
+log "Retrieving assets"
+while IFS=" " read -r filename size url; do
+    if ! [[ -f "${cache_dir}/${filename}" ]]; then
+        log "Downloading asset: ${filename} (${size} Bytes)"
+        curl -s -L --output "${cache_dir}/${filename}" "${url}"
+    fi
+    checksum=$(jq -r "[.[] | select(.filename == \"${filename}\")] | first | .checksum" <<< "${assets_data}")
+    echo "${checksum} ${cache_dir}/${filename}" | sha256sum --check -
+done < <(jq -r '.assets[] | "\(.name) \(.size) \(.browser_download_url)"' <<< "${assets_release_data}")
+
+log "Extracting assets"
+TEMP=$(mktemp -d) && trap "rm -rf ${TEMP}" EXIT || exit 255
+for ((i=$(jq length <<< "${assets_data}") - 1; i >= 0; --i)); do
+    IFS=" " read -r filename sourcedir targetdir \
```

Comment:
Do you need to set `$IFS` here?

Reply:
I've explicitly set it on all read commands in case the user has set it. I don't think it's really needed, but better safe than sorry. We can export it at the start of the script though, if you prefer it that way.

Reply:
Doesn't bash set it as `" \t\n"` at the start of the script automatically?

Reply:
Yep, not necessary and can be removed as it doesn't inherit the value. The bash man page is only talking about default values, so wasn't 100% sure.